File: producer/api/models/answer.js
const mongoose = require("mongoose");

const answerSchema = new mongoose.Schema({
    _id: mongoose.Schema.Types.ObjectId,
    question: {type: mongoose.Schema.Types.ObjectId, ref: "Question",required: true},
    response: {type: mongoose.Schema.Types.ObjectId, ref: "Response",required: true},
    text: {type: String}, 
})

module.exports = mongoose.model("Answer", answerSchema);

File: go-producer/models/answer.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type AnswerSchema struct {
	Id       primitive.ObjectID `json:"_id,omitempty"`
	Question primitive.ObjectID `json:"question,omitempty" validate:"required"`
	Response primitive.ObjectID `json:"response,omitempty" validate:"required"`
	Text     string             `json:"text,omitempty"`
}
